api/src/main/java/com/velocitypowered/api/event/command/PostCommandExecuteEvent.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2020-2023 Velocity Contributors
+ *
+ * The Velocity API is licensed under the terms of the MIT License. For more details,
+ * reference the LICENSE file in the api top-level directory.
+ */
+
+package com.velocitypowered.api.event.command;
+
+import com.google.common.base.Preconditions;
+import com.velocitypowered.api.command.CommandSource;
+
+/**
+ * This event is fired when velocity executed a command. This event is called after the event
+ * is handled. It's not fired for forwarded commands or commands with syntax errors.
+ */
+public final class PostCommandExecuteEvent {
+
+  private final CommandSource commandSource;
+  private final String command;
+  private final ExecuteResult result;
+
+  /**
+   * Constructs a CommandExecuteEvent.
+   *
+   * @param commandSource the source executing the command
+   * @param command the command being executed without first slash
+   * @param result the result of this command
+   */
+  public PostCommandExecuteEvent(CommandSource commandSource, String command,
+                                 ExecuteResult result) {
+    this.commandSource = Preconditions.checkNotNull(commandSource, "commandSource");
+    this.command = Preconditions.checkNotNull(command, "command");
+    this.result = Preconditions.checkNotNull(result, "result");
+  }
+
+  /**
+   * Get the source of this executed command.
+   *
+   * @return the source
+   */
+  public CommandSource getCommandSource() {
+    return commandSource;
+  }
+
+  /**
+   * Gets the original command line executed without the first slash.
+   *
+   * @return the original command
+   */
+  public String getCommand() {
+    return command;
+  }
+
+  /**
+   * Returns the result of the command execution.
+   *
+   * @return the execution result
+   */
+  public ExecuteResult getResult() {
+    return result;
+  }
+
+  @Override
+  public String toString() {
+    return "CommandExecuteEvent{"
+            + "commandSource=" + commandSource
+            + ", command=" + command
+            + '}';
+  }
+
+  /**
+   * The result of the executed command.
+   */
+  public enum ExecuteResult {
+    EXECUTED,
```

Comment:
I like the generality of this new approach (indeed, the next major version will likely replace the `boolean` return of the `execute*` family of methods by an enum similar to this one). For this reason, I would move this enum to its own file.

I agree with 4drian3d; even with this change I still don't like the name. As an API user I would expect this event to be fired only in the case that a command was actually executed. If the command input has incorrect syntax, then a command clearly wasn't executed. Similarly, if an exception occurs during command execution, the command wasn't fully executed and the `Post` part of the event/code flow wasn't reached. (The javadoc needs an update btw.)

I understand the need for an event to listen to everything command-related. However, this use case is very niche (I can only think about logging plugins that upload exceptions to e.g. services like Sentry---even that should be handled by a SLFJ integration). One approach would be to have one event type (`CommandSyntaxErrorEvent`, `CommandForwardedEvent`, etc.) per case, but this could be too "heavy"/"enterprisey". Another way to partially satisfy your use case follows. A niche feature provided by Brigadier is setting an execution callback (https://github.com/Mojang/brigadier/blob/a6c4baa1b639da62f7f0df62aa61de713e2136c8/src/main/java/com/mojang/brigadier/CommandDispatcher.java#L107 ). Brigadier calls this function with the results of a command execution, or on error. I say "partially" because your plugin wouldn't be notified of syntax errors, only of execution success, forward (by checking the `int` return code--this is API), or error.

In short, we need to come up with a better name for this event now that it is fired in response to more cases than execution success on the proxy.

Regarding not exposing the `ParseResults`: this is a public type provided by Brigadier, and in case the parse fails it contains data on why that is the case. Under this new approach, we could provide a nullable getter.

Reply:
> A mostly unknown feature provided by Brigadier is setting an execution callback 

This looks nice and for me the right thing, however this is not exposed in the Velocity API. It's possible to set **one** Consumer but not multiple so exposing it only makes sense as an event.

After your hint this should be the better idea. Instead of implementing the event like my suggestion simply use the consumer callback, call the event there and it's done.

Then I don't need to add more logic to VelocityCommandManager and it's done in a few lines of code.